sdk/typescript/src/client/ika-client.ts

```diff
@@ -138,6 +221,100 @@ export class IkaClient {
 		return this.cachedObjects!;
 	}
 
+	/**
+	 * Get all available network encryption keys.
+	 * This method fetches and caches all encryption keys for efficient access.
+	 *
+	 * @returns Promise resolving to an array of all network encryption keys
+	 * @throws {NetworkError} If the encryption keys cannot be fetched
+	 */
+	async getAllNetworkEncryptionKeys(): Promise<NetworkEncryptionKey[]> {
+		if (!this.cache) {
+			return this.#fetchEncryptionKeys();
+		}
+
+		if (this.cachedEncryptionKeys.size > 0) {
+			return Array.from(this.cachedEncryptionKeys.values());
+		}
+
+		if (this.encryptionKeysPromise) {
+			await this.encryptionKeysPromise;
+			return Array.from(this.cachedEncryptionKeys.values());
+		}
+
+		await this.#fetchEncryptionKeys();
+		return Array.from(this.cachedEncryptionKeys.values());
+	}
+
+	/**
+	 * Get the latest network encryption key.
+	 * This is the most recent encryption key created for the network.
+	 *
+	 * @returns Promise resolving to the latest network encryption key
+	 * @throws {NetworkError} If the encryption keys cannot be fetched
+	 */
+	async getLatestNetworkEncryptionKey(): Promise<NetworkEncryptionKey> {
```

Comment:
I'm not sure this logic works, but I am sure that it is inefficent. The public output can be big, and parsing is non-negligible, maybe try to understand what the latest is and get just that one